src/database/DataTypes/Item.js

```diff
@@ -224,6 +216,96 @@ export class Item extends Realm.Object {
   toString() {
     return `${this.code} - ${this.name}`;
   }
+
+  /**
+   * Finds all this items batches in a specific location.
+   * If no location is passed, all of this items ItemBatches
+   * are returned.
+   * @param {Location} location
+   */
+  getBatchesInLocation({ id } = {}) {
+    if (!id) return this.batches;
+    return this.batches.filtered('location.id = $0', id);
+  }
+
+  /**
+   * Returns the number of batches in a specific location. If
+   * no location is passed, finds the number of batches for all
+   * batches.
+   * @param {Location} location
+   */
+  getNumberOfBatches(location) {
+    return this.getBatchesInLocation(location).length;
+  }
+
+  /**
+   * Returns the sum of each ItemBatch.totalQuantity in a specific
+   * location. If no location is passed, finds the totalQuantity
+   * for all batches.
+   * @param {Location} location
+   */
+  getQuantityInLocation(location) {
+    return this.getBatchesInLocation(location).reduce(
+      (sum, { numberOfPacks = 0, packSize = 1 } = {}) => sum + numberOfPacks * packSize,
```

Comment:
itemBatch does have totalQuantity
can use getTotal utility method getTotal(this.getBatchesInLocation(location), 'totalQuantity').

or this.getBatchesInLocation(location).sum('numberOfPacks') <- we enforce packSize of 1 in mobile

Reply:
done d4e4cce3f84bf0e9a9e36da471d51ac2c18c6995